exporter/signalfxexporter/exporter.go

```diff
@@ -156,7 +155,9 @@ func (se *signalfxExporter) start(ctx context.Context, host component.Host) (err
 			IdleConnTimeout:     se.config.DimensionClient.IdleConnTimeout,
 			Timeout:             se.config.DimensionClient.Timeout,
 		})
-	dimClient.Start()
+	// Use background context to conform to spec's requirements for a component's
+	// long-running background processes.
+	dimClient.Start(context.Background())
```

Comment:
Why? This sound like a change in behavior and something that can actually introduce a leak

Reply:
Right, I would have thought it looked concerning as well, but the [spec for `Start`](https://pkg.go.dev/go.opentelemetry.io/collector/component#Component) says:
```
	// If the component needs to perform a long-running starting operation then it is recommended
	// that Start() returns quickly and the long-running operation is performed in background.
	// In that case make sure that the long-running operation does not use the context passed
	// to Start() function since that context will be cancelled soon and can abort the long-running
	// operation. Create a new context from the context.Background() for long-running operations.
```
The `dimClient` creates a context and cancel func from the passed in context, and ends up being a long-running process. It's safe since it will call cancel in its own `shutdown`, not relying on `context.Background` to be cancelled.

In reality, the context passed to start is **not** canceled right away and stays "alive" through the rest of the time the collector is running (so the spec doesn't line up with current behavior), but I assume there's some potential in future use-cases where it could be which is what the spec is trying to protect against.

Reply:
Ok, why the Start needs to take a context argument in that case?

Reply:
It doesn't need to, but it's personally clearer to me that a method call is more than just a simple short running synchronous method if the context is passed in. Seeing context passed in makes it more clear that that something more is going on. (Although the name `Start` is also pretty clear).

We can remove if you prefer.

Reply:
but we don't pass context to other start methods added in this PR

Reply:
Coming back to this, I was also concerned by the passing in of `context.Background` before looking at usage again.

I've updated to just directly create a context with cancel from `context.Background` in the dim client's start. 👍